testing/test_reports.py

```diff
@@ -305,13 +305,75 @@ def check_longrepr(longrepr):
 
         data = report._to_json()
         loaded_report = report_class._from_json(data)
+
+        assert loaded_report.failed
         check_longrepr(loaded_report.longrepr)
 
         # make sure we don't blow up on ``toterminal`` call; we don't test the actual output because it is very
         # brittle and hard to maintain, but we can assume it is correct because ``toterminal`` is already tested
         # elsewhere and we do check the contents of the longrepr object after loading it.
         loaded_report.longrepr.toterminal(tw_mock)
 
+    def test_chained_exceptions_no_reprcrash(
+        self, testdir, tw_mock,
+    ):
+        """Regression test for tracebacks without a reprcrash (#5971)
+
+        This happens notably on exceptions raised by multiprocess.pool: the exception transfer
+        from subprocess to main process creates an artificial exception which, ExceptionInfo
+        can't obtain the ReprFileLocation from.
+        """
+        testdir.makepyfile(
+            """
+            # equivalent of multiprocessing.pool.RemoteTraceback
+            class RemoteTraceback(Exception):
+                def __init__(self, tb):
+                    self.tb = tb
+                def __str__(self):
+                    return self.tb
+
+            def test_a():
+                try:
+                    raise ValueError('value error')
+                except ValueError as e:
+                    # equivalent to how multiprocessing.pool.rebuild_exc does it
```

Comment:
could you use `multiprocessing` to get the exact situation?  might be more representative / future proof

Reply:
Yeah I've thought about it and imagined it would incur too much overhead... but I just tried it and it is small (<0.4s) on Windows, so it will be much smaller on Linux, so making it more reliable is worth it I think.